File: src/main/java/com/starshootercity/abilities/FlightAllowingAbility.java
package com.starshootercity.abilities;

import net.kyori.adventure.util.TriState;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public interface FlightAllowingAbility extends Ability {
    boolean canFly(Player player);
    float getFlightSpeed(Player player);
    default @NotNull TriState getFlyingFallDamage(Player player) {
        return TriState.FALSE;
    }
}

File: core/src/main/kotlin/ru/turbovadim/abilities/FlightAllowingAbility.kt
package ru.turbovadim.abilities

import net.kyori.adventure.util.TriState
import org.bukkit.entity.Player

interface FlightAllowingAbility : Ability {
    fun canFly(player: Player): Boolean
    fun getFlightSpeed(player: Player): Float
    fun getFlyingFallDamage(player: Player): TriState {
        return TriState.FALSE
    }
}
